Updated workflow documentation (#28)
wso2-extensions/identity-tools-cli: docs/resource-propagation.md

# Resource Propagation across Multiple Environments

> This repository provides instructions on how to continuously propagate resources across multiple environments using the IAM-CTL tool.

Reviewer: ```suggestion
This repository provides instructions on continuously propagating resources across multiple environments using the IAM-CTL tool.
```

## Recommended Workflow

To manage resources when multiple environments exist, follow this recommended workflow:

> 1. **Set Up WSO2 Identity Server Instances**: Have separate WSO2 Identity Server instances for each environment, hosted somewhere accessible by the public.

Reviewer: ```suggestion
1. **Set up WSO2 Identity Server instances**: Have separate WSO2 Identity Server instances for each environment, hosted somewhere accessible by the public.
```

> 2. **Create Management Application**: Create an application with the Management Application enabled in the target Identity Server. Update the OAuth inbound authentication configuration with a dummy callback URL. Make note of the Client ID and Client Secret for later use. *(Automation will be implemented in the future)*

Reviewer: ```suggestion
2. **Create a management application**: Create an application with the Management Application capabilities enabled in the target Identity Server. Update the OAuth inbound authentication configuration with a dummy callback URL. Make note of the Client ID and Client Secret for later use. *(Automation will be implemented in the future)*
```

> 3. **Git Repository Setup**:

Reviewer: ```suggestion
3. **Git repository setup**:
```

   - Create a Git repository with branches dedicated to each environment. Use the environment name (same as the branch name) as an environment variable with the key `ENV` for each branch. This is where the resource config files will reside for that environment.

>    - **Manual Approach**:

Reviewer: ```suggestion
   - **Manual approach**:
```

     - Clone the repository to your local machine.
     - Download the latest release of the IAM-CTL tool from [GitHub Releases](https://github.com/wso2-extensions/identity-tools-cli/releases) and unzip it.
     - Run the following command, replacing `<path to the cloned repo folder>` with the appropriate path: 
       ```
       iamctl setupCLI -d "<path to the cloned repo folder>"
       ```
     - Create subfolders for each environment in the `config` folder and configure each `serverConfig.json` file with the relevant server details and other configurations.

>      - *Note: Variables of `serverConfig.json` can also be added as environment values. Refer to the [documentation](<LINK>) for more information on loading server configurations from environment variables.*

Reviewer: Add the link here

>      - *Note: Variables of `serverConfig.json` can also be added as environment values. Refer to the [documentation](<LINK>) for more information on loading server configurations from environment variables.*

Reviewer 2: ```suggestion
     - *Note: Variables of `serverConfig.json` can also be added as environment values. Refer to the [documentation](https://github.com/wso2-extensions/identity-tools-cli/blob/master/docs/cli-mode.md#load-server-configurations-from-environment-variables) for more information on loading server configurations from environment variables.*
```

>    - **Workflow Automation**:

Reviewer: ```suggestion
   - **Workflow automation**:
```

>      - Set up the environment variable. Refer to the [documentation](<LINK>) for more information on loading server configurations from environment variables.

Reviewer: Add link

>      - Set up the environment variable. Refer to the [documentation](<LINK>) for more information on loading server configurations from environment variables.

Reviewer 2: ```suggestion
     - Set up the environment variable. Refer to the [documentation](https://github.com/wso2-extensions/identity-tools-cli/blob/master/docs/cli-mode.md#load-server-configurations-from-environment-variables) for more information on loading server configurations from environment variables.
```

     - Add a GitHub Action with the custom action `@action/setup` (replace with the correct action reference).
     - Trigger the action.
     - An automatically generated `config` folder will be added to the repository.

> 4. **Export Resources**:

Reviewer: ```suggestion
4. **Export resources**:
```

>    - Export resources from the lower environment to the lower environment branch using another GitHub Action with the custom action `@action/export`.

Reviewer: ```suggestion
   - Export resources from the lower environment to the lower environment branch using another GitHub action with the custom action `@action/export`.
```

   - A pull request will be created with the changes. Review and verify the correctness of the propagation, and then merge the pull request.

> 5. **Environment-specific Variables**:

Reviewer: ```suggestion
5. **Environment-specific variables**:
```

   - If there are environment-specific variables, add keyword placeholders to the exported files and add the relevant keyword mapping to the tool configurations.

> 6. **Merge to Higher Environment Branch**:

Reviewer: ```suggestion
6. **Merge to higher environment branch**:
```

   - Merge the lower environment branch to the higher environment branch. *(Automation can be implemented to automatically merge the pull request from the lower environment to the higher environment branch.)*

> 7. **Import Resources**:

Reviewer: ```suggestion
7. **Import resources**:
```

   - Use the import command to deploy the resources from the repository to the higher environment using another GitHub Action with the custom action `@action/import`.
   - *(Automation can also be implemented by triggering the Git action automatically upon merging to the branch.)*

> 8. **Update and Import New Changes**:

Reviewer: ```suggestion
8. **Update and import new changes**:
```
